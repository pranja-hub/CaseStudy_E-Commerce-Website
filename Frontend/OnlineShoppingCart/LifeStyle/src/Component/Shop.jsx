import React from "react";
import "./StyleSheet.css";
import { Link, useNavigate } from "react-router-dom";

const Shop = () => {
  const nav = useNavigate();
  const Logout = () => {
    if (typeof window !== "undefined") {
      localStorage.removeItem("jwt");
      nav("/");
    }
  };

  return (
    <>
      <div>
        <nav className="navbar navbar-inverse bg-dark navbar-dark">
          <div className="container">
            <div className="navbar-header">
              <div className="navbar-brand active">
                Lifestyle Store
              </div>
            {localStorage.getItem("jwt") && (
              <ul class="nav navbar-nav navbar-right">
                <li>
                  <button
                    onClick={Logout}
                    style={{  cursor: "pointer",
                    backgroundColor: "floralwhite",
                    position: "absolute",
                    top: "14px",
                    right: "6em"}}
                    >
                    <span>
                    Logout
                  </span>
                  </button>
                </li>
              </ul>
            )}
          </div>
          </div>
        </nav>

        <div id="banner_image">
          <div className="container">
            <div className="banner_content">
              <Link to="/product">
              <button className="btn btn-danger btn-lg active"
              style={{
                marginTop: "3em",
                marginLeft: "15em"
              }}>
                Shop Now
              </button>
              </Link>
            </div>
          </div>
        </div>
        <div className="footer">
          <div className="container">
            <center>
              <h>
                <b>Copyright &#169; Lifestyle Store. All Rights Reserved</b> and{" "}
                <b>Contact Us: +91 90000 00000</b>
              </h>
            </center>
          </div>
        </div>
      </div>
    </>
  );
};

export default Shop;
