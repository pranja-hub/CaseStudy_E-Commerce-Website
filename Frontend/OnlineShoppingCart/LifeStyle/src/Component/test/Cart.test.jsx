import React from "react";
import renderer from "react-test-renderer";
import Shop from "../Shop";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  Link: () => mockedUsedNavigate,
}));

afterAll(() => {
  jest.clearAllMocks();
});

it("renders correctly", () => {
  const tree = renderer.create(<Shop />).toJSON();

  console.log(tree);

  expect(tree).toMatchSnapshot();
});
