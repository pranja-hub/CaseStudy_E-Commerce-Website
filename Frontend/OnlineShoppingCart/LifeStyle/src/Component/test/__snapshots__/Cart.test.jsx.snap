// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-inverse bg-dark navbar-dark"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <div
          className="navbar-brand active"
        >
          Lifestyle Store
        </div>
      </div>
    </div>
  </nav>
  <div
    id="banner_image"
  >
    <div
      className="container"
    >
      <div
        className="banner_content"
      />
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="container"
    >
      <center>
        <h>
          <b>
            Copyright Â© Lifestyle Store. All Rights Reserved
          </b>
           and
           
          <b>
            Contact Us: +91 90000 00000
          </b>
        </h>
      </center>
    </div>
  </div>
</div>
`;
